src/utils/globalAttributes.ts

```diff
@@ -0,0 +1,73 @@
+import { createObserver } from "./observers";
+
+type AttributeObject = { [k: string]: any };
+type AllowedGlobalAttribute = "lang";
+const allowedGlobalAttributes = ["lang"];
+
+const mutationObserverMap: WeakMap<GlobalAttrComponent, MutationObserver> = new WeakMap();
+
+/**
+ * Watches global attributes of a component.
+ *
+ * Derived from: https://gist.github.com/willmartian/b4dd6b57d71dd0438fb9e7c6f4048578
+ */
+export interface GlobalAttrComponent {
+  /**
+   * The host element.
+   */
+  readonly el: HTMLElement;
+
+  /**
+   * The watched attributes object.
+   * Should be stateful.
+   * '@State() inheritedAttributes = {};'
+   */
+  globalAttributes: AttributeObject;
+}
+
+/**
+ * Helper to set up listening for changes to global attributes.
+ *
+ * render() {
+ *   const lang = this.inheritedAttributes['lang'] ?? 'en';
+ *   return <div>My lang is {lang}</div>;
+ * }
+ */
+export function watchGlobalAttributes(component: GlobalAttrComponent, attributes: AllowedGlobalAttribute[]): void {
+  const attributeObject: AttributeObject = {};
+  const { el } = component;
+
+  const updateAttributesObject = () => {
+    attributes
+      .filter((attr) => !!allowedGlobalAttributes.includes(attr) && !!el.hasAttribute(attr))
+      .forEach((attr) => {
+        const value = el.getAttribute(attr);
+
+        if (value !== null) {
+          attributeObject[attr] = value;
+        }
+      });
+
+    component.globalAttributes = { ...attributeObject };
+  };
+
+  updateAttributesObject();
+
+  const mutationObserver = createObserver("mutation", () => updateAttributesObject());
+
+  mutationObserver.observe(el, {
+    attributeFilter: attributes
+  });
+
+  mutationObserverMap.set(component, mutationObserver);
```

Comment:
@jcfranco is this what you had in mind?

Reply:
Very close. Instead of a `component-mutation` observer map, I was thinking an `element-[component, attributes]` map (or 2 separate maps to avoid arrays) that a single observer and update function could use.

Reply:
@jcfranco is that because you think the `attributes` may change after initialization? I don't think that's a very likely case. Maybe we could defer doing that?

Reply:
> is that because you think the attributes may change after initialization?

No, it's to be able to use a single observer and associate each element to their watched attributes.

Reply:
@jcfranco I'm not following how you can use a single observer. Can you take that part? I'm not seeing how it can setup a single observer to call a generic method like that.

Reply:
Apologies for the delay on this. I've updated the PR with what I was envisioning and also added a spec test for the util. Can you take a look? 👀